src/App.js

```diff
@@ -1,65 +1,45 @@
 import React, { useEffect, useState } from 'react';
 import { withAuthenticator } from 'aws-amplify-react';
-import { API, graphqlOperation } from 'aws-amplify';
+import { connect } from 'react-redux';
 import Notes from './components/Notes';
 import Form from './components/Form';
-import { listNotes } from './graphql/queries';
-import { createNote, updateNote, deleteNote } from './graphql/mutations';
+import {
+  fetchNotesAction, addNotesAction, deleteNoteAction, updateNoteAction,
+} from './modules/notes/notesActions';
 
 import './App.css';
 
-
 const initialState = { id: null, name: '', description: '' };
-const initialNotes = { notes: [] };
 
-const App = () => {
+const App = ({
+  // eslint-disable-next-line no-shadow
+  notes, fetchNotesAction, addNotesAction, deleteNoteAction, updateNoteAction,
+}) => {
   const [formState, setFormState] = useState(initialState);
-  const [notesState, setNotes] = useState(initialNotes);
   const [edit, setEdit] = useState(false);
+  const { notesArray } = notes;
+
   const setInput = (key, value) => {
     setFormState({ ...formState, [key]: value });
   };
 
-  const fetchNotes = async () => {
-    try {
-      const { data: { listNotes: { items } } } = await API.graphql(graphqlOperation(listNotes));
-      setNotes({ notes: items });
-    } catch (err) {
-      // eslint-disable-next-line no-console
-      console.log('error fetching notes...', err);
-    }
-  };
-
   useEffect(() => {
-    fetchNotes();
+    fetchNotesAction();
   }, []);
 
   const onCreateNote = async () => {
     if (!formState.name || !formState.description) return;
     const note = { ...formState };
-    const newNotes = [note, ...notesState.notes];
-    setNotes({ notes: newNotes });
+    const newNotes = [note, ...notesArray];
     setFormState(initialState);
-    setEdit(false);
-    try {
-      await API.graphql(graphqlOperation(createNote, { input: note }));
-    } catch (err) {
-      // eslint-disable-next-line no-console
-      console.log('error creating todo', err);
-    }
+    addNotesAction(note, newNotes);
+    fetchNotesAction();
```

Comment:
Always doing a fetch is not efficient. The change will already have been made in redux so is it necessary?